src/main/java/mcp/mobius/waila/registry/Registrar.java

```diff
@@ -288,14 +288,16 @@ public void lock() {
         hash[1] = hash(blacklist.blockEntityTypes, BuiltInRegistries.BLOCK_ENTITY_TYPE);
         hash[2] = hash(blacklist.entityTypes, BuiltInRegistries.ENTITY_TYPE);
 
-        if (Waila.BLACKLIST_CONFIG.isFileExists() && !Arrays.equals(Waila.BLACKLIST_CONFIG.get().pluginHash, hash)) {
-            Waila.BLACKLIST_CONFIG.backup();
+        if (!Arrays.equals(Waila.BLACKLIST_CONFIG.get().pluginHash, hash)) {
+            if (Waila.BLACKLIST_CONFIG.isFileExists()) {
+                Waila.BLACKLIST_CONFIG.backup();
+            }
```

Comment:
Hmm... yeah I think the `isFileExists` check will always return true. Maybe store it in a variable first before trying to get the config.
```suggestion
        boolean doBackup = Waila.BLACKLIST_CONFIG.isFileExists();
        if (!Arrays.equals(Waila.BLACKLIST_CONFIG.get().pluginHash, hash)) {
            if (doBackup) {
                Waila.BLACKLIST_CONFIG.backup();
            }
```

Reply:
Doesn't the `ConfigIO` class already do the backup job? 🤔 In `read` method

Reply:
It's for when the file is an invalid JSON.

Reply:
Oh yes that's exactly what I did

Reply:
Nope no luck. Same issue.